internal/validator/rule_validator.go

```diff
@@ -0,0 +1,39 @@
+package validator
+
+import (
+	"github.com/meshplus/bitxhub/internal/ledger"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	FabricRuleAddr = "0x00000000000000000000000000000000000000a0"
+)
+
+// Validator is the instance that can use wasm to verify transaction validity
+type ValidationEngine struct {
+	ledger ledger.Ledger
+	logger logrus.FieldLogger
+}
+
+// New a validator instance
+func NewValidationEngine(ledger ledger.Ledger, logger logrus.FieldLogger) *ValidationEngine {
+	return &ValidationEngine{
+		ledger: ledger,
+		logger: logger,
+	}
+}
+
+// Verify will check whether the transaction info is valid
+func (ve *ValidationEngine) Validate(address, from string, proof []byte, validators string) (bool, error) {
+	vlt := ve.getValidator(address)
```

Comment:
Will it get a `nil` validator?

Reply:
It will return wasm validator for default.

Reply:
What if the address doesn't exist in `wasm`?

Reply:
The error will throw out in Verify(...). The wasm instance will be initialized in Verify(...).